python/pyspark/ml/feature.py

```diff
@@ -755,6 +759,29 @@ class StringIndexerModel(JavaModel):
     Model fitted by StringIndexer.
     """
 
+    def invert(self, inputCol, outputCol):
+        """
+        Return a model to perform the inverse transformation.
+        Note: By default we keep the original columns during this transformation, so the inverse
+        should only be used on new columns such as predicted labels.
+        """
+        return StringIndexerInverse(self._java_obj.invert(inputCol, outputCol))
+
+
+class StringIndexerInverse(JavaTransformer):
+    """
+    Transform a provided column back to the original input types using the metadata on
+    the input column.
+    Note: By default we keep the original columns during StringIndexerModel's transformation,
+    so the inverse should only be used on new columns such as predicted labels.
+    """
+
+    def __init__(self, java_obj):
```

Comment:
I'd prefer this stick to the usual setup, rather than only being created via StringIndexerModel.invert.  We should have a regular constructor, not this specialized one; StringIndexerModel.invert can do something equivalent, but manually.